community/kernel/src/main/java/org/neo4j/kernel/impl/api/DefaultTransactionTracer.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.api;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.neo4j.helpers.Clock;
+import org.neo4j.kernel.impl.transaction.tracing.CommitEvent;
+import org.neo4j.kernel.impl.transaction.tracing.LogAppendEvent;
+import org.neo4j.kernel.impl.transaction.tracing.LogForceEvent;
+import org.neo4j.kernel.impl.transaction.tracing.LogForceWaitEvent;
+import org.neo4j.kernel.impl.transaction.tracing.LogRotateEvent;
+import org.neo4j.kernel.impl.transaction.tracing.SerializeTransactionEvent;
+import org.neo4j.kernel.impl.transaction.tracing.StoreApplyEvent;
+import org.neo4j.kernel.impl.transaction.tracing.TransactionEvent;
+import org.neo4j.kernel.impl.transaction.tracing.TransactionTracer;
+
+public class DefaultTransactionTracer implements TransactionTracer, LogRotationMonitor
+{
+    private final Clock clock;
+    private final AtomicLong counter = new AtomicLong();
+    private final AtomicLong accumulatedTotalTimeNanos = new AtomicLong();
+
+    private long startTimeNanos;
```

Comment:
It is the same thread creating the `LogAppendEvent` and the `LogRotateEvent` and closing both. Is the `volatile` really needed?